src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -550,37 +1186,76 @@ POTRF_BATCH_LAUNCHER_USM(std::complex<double>, cusolverDnZpotrfBatched)
 
 #undef POTRF_BATCH_LAUNCHER_USM
 
-sycl::event potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, std::int64_t n,
-                        std::int64_t nrhs, float *a, std::int64_t lda, std::int64_t stride_a,
-                        float *b, std::int64_t ldb, std::int64_t stride_b, std::int64_t batch_size,
-                        float *scratchpad, std::int64_t scratchpad_size,
-                        const std::vector<sycl::event> &dependencies) {
-    throw unimplemented("lapack", "potrs_batch");
-}
-sycl::event potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, std::int64_t n,
-                        std::int64_t nrhs, double *a, std::int64_t lda, std::int64_t stride_a,
-                        double *b, std::int64_t ldb, std::int64_t stride_b, std::int64_t batch_size,
-                        double *scratchpad, std::int64_t scratchpad_size,
-                        const std::vector<sycl::event> &dependencies) {
-    throw unimplemented("lapack", "potrs_batch");
-}
-sycl::event potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, std::int64_t n,
-                        std::int64_t nrhs, std::complex<float> *a, std::int64_t lda,
-                        std::int64_t stride_a, std::complex<float> *b, std::int64_t ldb,
-                        std::int64_t stride_b, std::int64_t batch_size,
-                        std::complex<float> *scratchpad, std::int64_t scratchpad_size,
-                        const std::vector<sycl::event> &dependencies) {
-    throw unimplemented("lapack", "potrs_batch");
-}
-sycl::event potrs_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, std::int64_t n,
-                        std::int64_t nrhs, std::complex<double> *a, std::int64_t lda,
-                        std::int64_t stride_a, std::complex<double> *b, std::int64_t ldb,
-                        std::int64_t stride_b, std::int64_t batch_size,
-                        std::complex<double> *scratchpad, std::int64_t scratchpad_size,
-                        const std::vector<sycl::event> &dependencies) {
-    throw unimplemented("lapack", "potrs_batch");
+template <typename Func, typename T>
+inline sycl::event potrs_batch(const char *func_name, Func func, sycl::queue &queue,
+                               oneapi::mkl::uplo uplo, std::int64_t n, std::int64_t nrhs, T *a,
+                               std::int64_t lda, std::int64_t stride_a, T *b, std::int64_t ldb,
+                               std::int64_t stride_b, std::int64_t batch_size, T *scratchpad,
+                               std::int64_t scratchpad_size,
+                               const std::vector<sycl::event> &dependencies) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+
+    overflow_check(n, nrhs, lda, ldb, stride_a, stride_b, batch_size, scratchpad_size);
+
+    // cusolver function only supports nrhs = 1
+    if (nrhs != 1)
+        throw unimplemented("lapack", "potrs_batch", "cusolver potrs_batch only supports nrhs = 1");
```

Comment:
if at some point the support will be expanded, is there any way to bypass this check?

Reply:
There is no way to check if cuSolver supports this nrhs > 1 case. If support is expanded this check will have to be modified or deleted depending on the circumstances at the time.

Reply:
Could it be some single build-time constant, or preprocessor macro, switching it in single place. Like `static constexpr bool multiple_nrhs_support = false;` ?

Reply:
Okay, I have wrapped the check with a macros preprocessor, so if the user defines `POTRS_BATCHED_MULTIPLE_NRHS_SUPPORTED` then the check will be removed.